app/controllers/admin/projects_controller.rb

```diff
@@ -58,7 +53,7 @@ def filter
 
   def managed_update
     @project = Project.find(params[:id])
-    redirect_if_not_owner_or_admin(@project)
+    authorize! :managed_update, @project
 
```

Comment:
This is the more bespoke way of handling authorization when your controllers get a bit messier.